File: scripts/embed.js
import dotenv from "dotenv"
dotenv.config({ path: ".env.local" })

console.log(process.env.OPENAI_API_KEY)

import fs from "fs"
import OpenAI from "openai"

const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY
})

const quran = JSON.parse(
  fs.readFileSync("data/quran.json")
)

async function run() {

  for (const ayah of quran) {

    const text = ayah.arabic + " " + ayah.thai

    const res = await openai.embeddings.create({
      model: "text-embedding-3-small",
      input: text
    })

    ayah.embedding = res.data[0].embedding
  }

  fs.writeFileSync(
    "data/quran-embeddings.json",
    JSON.stringify(quran)
  )

  console.log("Embeddings created")
}

run()

File: scripts/embed.py
from sentence_transformers import SentenceTransformer
import json

model = SentenceTransformer("all-MiniLM-L6-v2")

with open("../data/quran.json", "r", encoding="utf-8") as f:
    ayat = json.load(f)

for a in ayat:
    text = a["arabic"] + " " + a["thai"]
    embedding = model.encode(text).tolist()
    a["embedding"] = embedding

with open("../data/embeddings.json", "w", encoding="utf-8") as f:
    json.dump(ayat, f, ensure_ascii=False, indent=2)

print("Embeddings created successfully")